exercises/concept-wip/arys-amazing-lasagna/.docs/introduction.md

```diff
@@ -0,0 +1,171 @@
+# Basics
+
+## Getting started
```

Comment:
Something that is necessary to solve the exercise, but I didn't see in the exercise's introduction:
- Mention implicit returns. Most languages require explicit returns with the `return` keyword, so Crystal doing implicit returns of the last expression of a method is something worth mentioning.
- Mention basic numerical operators. Plus and minus might be obvious, but not every language supports `*` for multiplication.

Reply:
I actually explained return but Erik suggested that piece to be removed.
I could add it back but a shorter version I guess.
And I can give a quick explanation of number operators.

Reply:
Both can be covered by adding one sentence each, e.g.:

> Methods always implicitly return the value of the last expression in the method.

and 

> Common number operators are: `+` (addition), `-` (subtraction), and `*` (multiplication).

I agree with Erik's point, but I would shorten the introduction by reducing the repetition and explicitness of the current explanations, for example:

> A variable can be declared and then redefine later.
> (...)
> The difference between constants and variables is that constants can't be redefined.

This is common knowledge in programming, those sentences can be removed.

Similarly the code example of creating a constant and what happens when you try to reassign it could be merged into a single example:

**Before**

~~~
```crystal
NUMBER = 1
puts NUMBER # => 1
```

Assigning a value to a constant that is already defined will result in a compile-time error.

```crystal
NUMBER = 1

NUMBER = 2
```

```console
In test.cr:3:1

 3 | NUMBER = 2
     ^
Error: already initialized constant NUMBER
```
~~~

**After**

~~~
Assigning a value to a constant that is already defined will result in a compile-time error.

```crystal
NUMBER = 1

NUMBER = 2
# Error: already initialized constant NUMBER
```
~~~